pkg/controller/routes.go

```diff
@@ -58,6 +60,8 @@ func routes(router *router.Router, registryTransport http.RoundTripper, recorder
 	appRouter := router.Type(&v1.AppInstance{}).Middleware(appdefinition.RequireNamespace, appdefinition.IgnoreTerminatingNamespace, appdefinition.FilterLabelsAndAnnotationsConfig)
 	appRouter.HandlerFunc(defaults.Calculate)
 	appRouter.HandlerFunc(scheduling.Calculate)
+	appRouter.HandlerFunc(quota.EnsureQuotaRequest)
```

Comment:
The problem with this approach is that, the quota object will always be created after all the handlers are processed. This means that `quota.WaitForAllocation` is always going to fail for the first several times since quota object is not created...

Reply:
Is the quota going to be effective even if it has to be create in the end? 

Now I am thinking an app could possibly escape the quota because the quota are not created at first?

Reply:
Left a comment above (I believe) addressing these concerns,